### PR TITLE
Improve `calc` and xilem_web's `mathml_svg` example

### PR DESCRIPTION
**Pull request by Philipp-M**
Reduces a little bit of the boilerplate in `calc`

### CODE REVIEW

File: xilem/examples/calc.rs [hunk 1]
```diff
@@ -187,6 +187,12 @@ impl Calculator {
 const DISPLAY_FONT_SIZE: f32 = 30.;
 const GRID_GAP: f64 = 2.;
 fn app_logic(data: &mut Calculator) -> impl WidgetView<Calculator> {
+    let num_row = |nums: [&'static str; 3], operator| {
+        flex_row((
+            nums.map(|num| digit_button(num).flex(1.)),
```
Comment (Philipp-M) [inline xilem/examples/calc.rs:192]: This is slightly more complex to avoid 2 additional lines, I would in practice rather use a little bit of repetition here, but I think it shows another interesting aspect, when it's not just an array with 3 elements...

File: xilem_web/web_examples/mathml_svg/src/main.rs
Comment (DJMcNab) [inline xilem_web/web_examples/mathml_svg/src/main.rs:1]: This *probably* should be in a different PR...

File: xilem/examples/calc.rs (hunk 1 above)
Comment (DJMcNab) [reply xilem/examples/calc.rs:192]: I think this change should use a different free component, since this doesn't actually use any state from within the function justifying being a closure.

**Review by DJMcNab — APPROVED** [review]
Thanks.

File: xilem/examples/calc.rs (hunk 1 above)
Comment (Philipp-M) [reply xilem/examples/calc.rs:192]: I thought about it, but than it adds more boilerplate to add all the typing, do you think that's worth it anyway?

File: xilem_web/web_examples/mathml_svg/src/main.rs
Comment (Philipp-M) [reply xilem_web/web_examples/mathml_svg/src/main.rs:1]: Should I extract this as a different PR? Or isn't this worth it?

**Comment by PoignardAzur** [comment]
This PR appears to have added a regression on my machine. Here's what my calc example looks like:

image
